### PR TITLE
Add support for additional external datastores

### PR DESCRIPTION
**Pull request by git-ival**
Closes rancher/qa-tasks#485 

Example database variables for postgres:
```
db_name                = "mydb"
db_engine              = "postgres"
db_engine_version      = "14.2"
db_password            = "<desired password>"
db_skip_final_snapshot = true
```

Example database variables for mariadb:
```
db_name                = "mydb"
db_engine              = "mariadb"
db_engine_version      = "10.5"
db_password            = "<desired password>"
db_skip_final_snapshot = true
```

### CODE REVIEW

File: control-plane/main.tf [hunk 1]
```diff
@@ -190,16 +191,16 @@ module "install_common" {
...
-  letsencrypt_email              = var.letsencrypt_email
-  rancher_image                  = var.rancher_image
-  rancher_image_tag              = var.rancher_image_tag
-  rancher_password               = var.rancher_password
-  use_new_bootstrap              = local.use_new_bootstrap
-  rancher_node_count             = var.rancher_node_count
-  byo_certs_bucket_path          = var.byo_certs_bucket_path
-  private_ca_file                = var.private_ca_file
-  cattle_prometheus_metrics      = var.cattle_prometheus_metrics
+  # helm_rancher_chart_values_path = "${path.module}/files/values/rancher_chart_values.tftpl"
```
Comment (floatingman) [inline control-plane/main.tf:194]: Why is this commented out?

**Review by floatingman — CHANGES_REQUESTED** [review]

File: control-plane/main.tf (hunk 1 above)
Comment (OmnesPotens) [reply control-plane/main.tf:194]: This is now handled within the `install_common` module itself, so we are using the default `helm_rancher_chart_values_path` that is supplied by the module

Comment (git-ival) [reply control-plane/main.tf:194]: We are using the default `helm_rancher_chart_values_path` which is supplied by the `install_common` module now. But this comment can probably be deleted

Comment (floatingman) [reply control-plane/main.tf:194]: No worries, just curious.

**Review by floatingman — APPROVED** [review]
Works for me

**Pull request merged**